### DISCUSSION TITLE
How to set privileges for Cancel action

### DISCUSSION THREAD

**Discussion by vovpff** [YetiForceCompany/YetiForceCRM#14445] [How to] [answered]
I found what all my users has ability to cancel record

![image](https://user-images.githubusercontent.com/8545791/112109062-2116f100-8bc2-11eb-8fd9-c4cc29a02442.png)

@mariuszkrzaczkowski How to set privileges for Cancel action?

This action also available and works if user role don't have permissions to change record status to Cancelled

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
from this mechanism there is table s_yf_record_quick_changer
```php
$rows = [
	[
		'tabid' => 90,
		'conditions' => App\Json::encode(['ssingleorders_status' => 'PLL_CANCELLED']),
		'values' => App\Json::encode(['ssingleorders_status' => 'PLL_ACCEPTED']),
		'btn_name' => 'PLL_ACCEPTED',
		'class' => 'btn-outline-success',
		'icon' => 'fas fa-check-double',
	],
	[
		'tabid' => 90,
		'conditions' => App\Json::encode(['ssingleorders_status' => 'PLL_CANCELLED']),
		'values' => App\Json::encode(['ssingleorders_status' => 'PLL_IN_REALIZATION']),
		'btn_name' => 'PLL_IN_REALIZATION',
		'class' => 'btn-outline-info',
		'icon' => 'far fa-play-circle',
	],
	[
		'tabid' => 90,
		'conditions' => App\Json::encode(['ssingleorders_status' => 'PLL_CANCELLED']),
		'values' => App\Json::encode(['ssingleorders_status' => 'PLL_FOR_VERIFICATION']),
		'btn_name' => 'PLL_FOR_VERIFICATION',
		'class' => 'btn-outline-info',
		'icon' => 'fas fa-user-check',
	],
];
$dbCommand = \App\Db::getInstance()->createCommand();
foreach ($rows as $row) {
	$dbCommand->insert('s_#__record_quick_changer', $row)->execute();
}
```

**Reply by vovpff** [reply]
@mariuszkrzaczkowski cool! Are there some GUI available for this feature?

**Reply by mariuszkrzaczkowski** [reply]
no

**Reply by mariuszkrzaczkowski** [reply]
https://gitdeveloper.yetiforce.com/index.php?module=SSingleOrders&view=Detail&record=3337&parent=118&mid=124
![image](https://user-images.githubusercontent.com/10198654/112123609-b1a10180-8bc1-11eb-9c99-4754cc3f76b5.png)

**Reply by vovpff** [reply]
@mariuszkrzaczkowski cool.

Could you provide `values` data of s_yf_record_quick_changer for this buttons? 

![image](https://user-images.githubusercontent.com/8545791/112124917-ed44c700-8bd3-11eb-8e5d-c49f4d2238cd.png)

**Reply by vovpff** [reply]
@mariuszkrzaczkowski thank you!

What about roles or profiles or user permissions?
Is it possible to use more than one condition? More than one value?

**Reply by mariuszkrzaczkowski** [reply]
`What about roles or profiles or user permissions?`
based on the permission to edit the field, you won't see the button without it

`Is it possible to use more than one condition? `
yes

`More than one value?`
yes

**Reply by vovpff** [reply]
> `What about roles or profiles or user permissions?`
> based on the permission to edit the field, you won't see the button without it

Ok 

> `Is it possible to use more than one condition? `
> yes
> 
> `More than one value?`
> yes

How to?

**Reply by mariuszkrzaczkowski** [reply]
think a little :P
['ssingleorders_status' => 'PLL_IN_REALIZATION', 'ccc' => 'vvvv', 'aaa' => 'vvvv', 'bbbb' => 'vvvv']

**Reply by vovpff** [reply]
> think a little :P
> ['ssingleorders_status' => 'PLL_IN_REALIZATION', 'ccc' => 'vvvv', 'aaa' => 'vvvv', 'bbbb' => 'vvvv']

I think about that but master answer will be more helpfull

**Comment by rioshaz** [comment]
how do u get these 3 buttons, double tick, play and review,, what they do ?
